File: java/sieve/src/main/java/Sieve.java
import java.util.List;
import java.util.ArrayList;
import java.util.Map;
import java.util.HashMap;

class Sieve {
    private final List<Integer> primes = new ArrayList<Integer>();

    // flag composite numbers with map 
    private final Map<Integer, Boolean> nonPrimes = new HashMap<Integer, Boolean>();

    Sieve(int maxPrime) {
        // increment until find next available unflagged number
        for (int i = 2; i <= maxPrime; i++) {
            if (!nonPrimes.containsKey(i)) {
                // first unflagged number is prime
                primes.add(i);
                // mark all multiples as not prime
                for (int j = i + i; j <= maxPrime; j += i) {
                    nonPrimes.put(j, true);
                }
            }
        }
    }

    List<Integer> getPrimes() {
        return this.primes;
    }
}

File: swift/sieve/Sources/Sieve/Sieve.swift
class Sieve {
    var primes = [Int]()
    init(_ n: Int) {
        var composites = Set<Int>()
        for i in 2...n {
            if (!composites.contains(i)) {
                primes.append(i)
                // mark multiples of i as composite
                var j = i * i
                while j <= n {
                    composites.insert(j)
                    j += i
                }
            }
        }
    }
}
